Database connections and models

qari7 opened the pull request:
- Added db connection in Backend/db.js

- Added data models:
    - story.js
    - agent.js
    - user.js
    - vote.js
    - critique.js
With the final two, I anticipate updates once I have an idea what data the process with generate from the backend

- Added mongoose dependency

qari7 committed:
establishing the db connection

qari7 committed:
moved it to right loc

qari7 committed:
made new user model

qari7 committed:
user data model

qari7 committed:
added more data models

qari7 committed:
agent model

qari7 committed:
added db connection export in server.js

qari7 committed:
added mongoose depency

qari7 committed:
Updated Backend Coverage Report

qari7 committed:
renaming

qari7 committed:
Merge branch 'db_init' of https://github.com/Capstone-Projects-2025-Spring/project-003-story-builder-team-1 into db_init

qari7 committed:
revert name change

qari7 committed:
Updated Backend Coverage Report

EdJuanGonzalez reviewed (commented):
In agent.js, this may need an id field after the lang graph stuff is implemented. For now it's good just keep that in mind. 

In story.js, where is prompt_id coming from? We can add the prompt to the database but it doesn't have it's own id. Unless that id comes from the database. 

In user.js, we decided against emails so that does not need to be there. Rahad mentioned that accounts is not something to really focus on.

In vote.js, what exactly is happening other than connecting to a database?

I don't think we should have the db url directly in the code. I think using a .env file would be best. 

Lastly, I currently have the account endpoints return a user_id to the frontend that it will store. If this needs to be the username, that is fine. I just need to send back a unique identifier that the frontend can send to the backend to use to edit the database. If a user_id is not the solution let me know and I will change it to username or whatever the user's unique identifier is.

qari7 committed:
changes column names and object types. removed email. fixed vote.js

qari7 commented:
- Regarding the id for agent, each collection has a built in _id key with a distinct id for each document (row).
- prompt_id was a mistake in naming. I renamed it to prompt.
- Removed email in user.js
- For vote.js, I had transferred the data model for vote incorrectly from the previous branch I was working on. The database connection is stored in Backed/db.js. I removed vote.js since the voting information can be represented in story.js for now.
- made an env file in Backend

qari7 committed:
added .env file

qari7 committed:
removed env

EdJuanGonzalez committed:
Merge branch 'main' into db_init

EdJuanGonzalez committed:
Updated Backend Coverage Report
